stress/cpuset_manager.go

```diff
@@ -78,7 +81,28 @@ func (manager *cpuStressserManager) TrackInjectorCores(cgroup cgroup.Manager) (c
 		return cpuset.NewCPUSet(), fmt.Errorf("error parsing cpuset allocated cores: %w", err)
 	}
 
-	manager.setCoresToBeStressed(manager.coresToBeStressed.Union(cores))
+	coresToBeStressedCount, _ := manager.userRequestedCores(cores.Size(), userRequestCount)
+	coresToBeStressed := cpuset.NewCPUSet(cores.ToSlice()[:coresToBeStressedCount]...)
+
+	manager.log.Infof("retrieved coresToBeStressed after applying userRequestCount %s: %s", userRequestCount.StrVal, coresToBeStressed)
+	manager.setCoresToBeStressed(manager.coresToBeStressed.Union(coresToBeStressed))
 
 	return cores, nil
```

Comment:
Shouldn't this now be `coresToBeStressed`?
```suggestion
	return coresToBeStressed, nil
```

Reply:
yes, good spot!